Code/Framework/AzToolsFramework/Tests/Prefab/Overrides/PrefabInspectorOverrideTests.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/Prefab/DocumentPropertyEditor/PrefabPropertyEditorNodes.h>
+#include <AzToolsFramework/UI/PropertyEditor/ComponentEditor.hxx>
+#include <Prefab/Overrides/PrefabInspectorOverrideTestFixture.h>
+
+namespace UnitTest
+{
+    using PrefabInspectorOverrideTest = PrefabInspectorOverrideTestFixture;
+
+    // These paths depends on multiple factors like the data in the component, how its reflected to serialize and edit contexts,
+    // how different DPE adapters likes ReflectionAdapter and PrefabAdapter construct the DPE DOM etc. Therefore, these may
+    // change in the future if the data stored in the DPE DOM itself changes and need to be modified accordingly to prevent test failures.
+    constexpr AZStd::string_view domPathToTranslateProperty = "/0/3/2";
+    constexpr AZStd::string_view domPathToOverriddenTranslateProperty = "/0/3/3";
+
+    TEST_F(PrefabInspectorOverrideTest, ValidatePresenceOfOverrideProperty)
+    {
+        AZ::EntityId newEntityId, parentContainerId, grandparentContainerId;
+        CreateEntityInNestedPrefab(newEntityId, parentContainerId, grandparentContainerId);
+
+        CreateAndValidateEditEntityOverride(newEntityId, grandparentContainerId);
+
+        GenerateComponentAdapterDoms(newEntityId);
+
+        AzToolsFramework::ComponentEditor::GetComponentAdapterContentsCallback callback = [](const AZ::Dom::Value& adapterContents)
+        {
+            EXPECT_FALSE(adapterContents.IsArrayEmpty());
+
+            AZ::Dom::Path pathToTranslateProperty(domPathToOverriddenTranslateProperty);
+            [[maybe_unused]] AZ::Dom::Value translateRow = adapterContents[pathToTranslateProperty];
+            EXPECT_EQ(translateRow.GetType(), AZ::Dom::Type::Node);
+            EXPECT_EQ(translateRow.ArraySize(), 3);
+
+            // First comes the PrefabOverrideProperty which will be used later by the PrefabAdapter to put up the PrefabOverrideIcon.
+            EXPECT_EQ(
+                translateRow[0][AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Type.GetName()].GetString(),
+                AzToolsFramework::Prefab::PrefabPropertyEditorNodes::PrefabOverrideProperty::Name);
+
+            // Then comes the label of the row.
+            EXPECT_EQ(translateRow[1][AZ::DocumentPropertyEditor::Nodes::Label::Value.GetName()].GetString(), "Translate");
+
+            // Then comes the actual property editor of the translate vector.
+            AZ::Dom::Value translateVectorInDom = translateRow[2][AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Value.GetName()];
+            EXPECT_EQ(translateVectorInDom[0].GetDouble(), 10.0);
```

Comment:
suggestion: There is definitely a balance with this but sometimes having fewer expectations/assertions per test can make failures a lot clearer (ideally we want there to be one reason and one reason only for a test to fail). This is a guideline and not a rule and doing what you've done here might make sense but now there are a few different ways the test could fail and it may make diagnosing failures more difficult. I just wanted to mention it for reference as sometimes having a few more tests and refactoring to avoid some duplication can be really useful long-term.

Reply:
I agree with the sentiment of having a test responsible for a single validation instead of a bunch of different once. However, what's happening here is that we are validating data in a DOM. So, if we have a structure like

- TopNode
  - Child1
     - GrandChild1
      - GrandChild2
        - property1 : true
        
We are basically going down this dom data and validating it at each stage. It makes sense to keep these expectations in one test in this case. Each dom is different and requires different validations. Breaking these expectations into multiple tests would make these much harder to write an manage because of the nature of these tests.

Reply:
Okay understood, it might be nice to look at some of the test matcher options to see if any of this could be streamlines but it's not a huge deal. You're right if the checks are all consistent/coherent and related that's definitely not as bad and makes sense 👍